hal-core/src/mem/mod.rs

```diff
@@ -112,3 +127,19 @@ impl RegionKind {
     /// Memory used for a page table.
     pub const PAGE_TABLE: Self = Self(KindInner::PageTable);
 }
+
+impl fmt::Debug for RegionKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
```

Comment:
do you know how mmio shows up here? is this not a concern of the memory map at this point, pending further bootstrap, or is  it `USED`?

Reply:
i _think_ they will show up as `Reserved` in the bootloader crate's memory map: https://github.com/rust-osdev/bootloader/blob/90f5b8910d146d6d489b70a6341d778253663cfa/src/bootinfo/memory_map.rs#L174-L175
but it isn't particularly clear from the comment. would have to actually look at the source to figure that out.

i think we'll want to get most mmio ranges from ACPI or BDA (does anyhting use that besides COM ports?) though.